### PR TITLE
Added is_null, is_empty, and is_null_or_empty check functions

### LINKED ISSUE #965: [FEATURE]: Create is_null, is_empty and is_null_or_empty checks
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Problem statement

Sometimes, we need specific columns to be null or empty to meet certain business rules. The only way DQX could let us do it would be with the is_equal_to check, but the argument "value" doesn't support None type. The set of checks is_not_null,_is_not_empty and is_not_null_or_empty doesn't have the "negate" argument.

Image

Image

### Proposed Solution

Create a set of checks called is_null, is_empty, and is_null_or_empty to allow checks that confirm whether a given column is null, empty, or both.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ashwin-911**
Implements #965 by adding three new check functions that verify columns ARE null, empty, or null/empty. These complement the existing is_not_null, is_not_empty, and is_not_null_and_not_empty checks.

- **is_null**: Checks if column values are null
- **is_empty**: Checks if column values are empty strings
- **is_null_or_empty**: Checks if column values are null or empty

All functions follow the established DQX pattern with proper decorators, docstrings, and comprehensive integration tests.

## Changes

- Added 3 new check functions to [src/databricks/labs/dqx/check_funcs.py](cci:7://file:///Users/ashwinsahay/dqx/src/databricks/labs/dqx/check_funcs.py:0:0-0:0) (53 lines)
- Added 3 comprehensive integration tests to [tests/integration/test_row_checks.py](cci:7://file:///Users/ashwinsahay/dqx/tests/integration/test_row_checks.py:0:0-0:0) (136 lines)
- All functions use `@register_rule("row")` decorator
- Google-style docstrings included
- Error messages follow DQX conventions

Resolves #965

## Tests

- [x] Code formatting: **10/10 score** (`make fmt`)
- [x] Unit tests: **509 passed** (`make test`)
- [x] Integration tests added (will run in CI/CD with Databricks environment)
- [x] All functions tested with multiple data types (string, int, map, array)
- [x] Edge cases covered (null, empty, whitespace)

### CODE REVIEW

**Review by ghanse — CHANGES_REQUESTED** [review]
Looking good so far. Can you update the documentation in `/reference/quality_checks.mdx` and add some performance tests?

File: docs/dqx/docs/reference/quality_checks.mdx [hunk 1]
```diff
@@ -107,13 +110,35 @@ For brevity, the `name` field in the examples is omitted and it will be auto-gen
...
+    arguments:
+      column: col1
+
+# is_not_null_and_not_empty check
+- criticality: error
+  check:
+    function: is_not_null_and_not_empty
+    arguments:
+      column: col1
+      trim_strings: true
```
Comment (mwojtyczka) [inline docs/dqx/docs/reference/quality_checks.mdx:149]:
```suggestion
```

File: src/databricks/labs/dqx/check_funcs.py [hunk 2]
```diff
@@ -164,6 +164,59 @@ def is_not_null(column: str | Column) -> Column:
...
+        Column object for condition
+    """
+    col_str_norm, col_expr_str, col_expr = get_normalized_column_and_expr(column)
+    return make_condition(
+        col_expr.isNotNull(), f"Column '{col_expr_str}' value is not null", f"{col_str_norm}_is_not_null"
+    )
+
+
+@register_rule("row")
+def is_empty(column: str | Column) -> Column:
```
Comment (mwojtyczka) [inline src/databricks/labs/dqx/check_funcs.py]: For consistency with other checks we should also add `trim_strings`. I can also see that we are missing it from `is_not_empty`

**Review by mwojtyczka — CHANGES_REQUESTED** [review]

File: docs/dqx/docs/reference/quality_checks.mdx [hunk 3]
```diff
@@ -28,8 +28,11 @@ You can also define your own custom checks in Python (see [Creating custom check
...
+| `is_empty`                         | Checks whether the values in the input column are empty.                                                                                                                                                                                                                                                                                                                                                                          | `column`: column to check (can be a string column name or a column expression)                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (mwojtyczka) [inline docs/dqx/docs/reference/quality_checks.mdx]: missing `trim_strings`

File: docs/dqx/docs/reference/quality_checks.mdx [hunk 4]
```diff
@@ -28,8 +28,11 @@ You can also define your own custom checks in Python (see [Creating custom check
...
+| `is_null_or_empty`                 | Checks whether the values in the input column are either null or empty.                                                                                                                                                                                                                                                                                                                                                           | `column`: column to check (can be a string column name or a column expression)                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (mwojtyczka) [inline docs/dqx/docs/reference/quality_checks.mdx]: missing `trim_strings`

File: docs/dqx/docs/reference/quality_checks.mdx [hunk 5]
```diff
@@ -28,8 +28,11 @@ You can also define your own custom checks in Python (see [Creating custom check
...
 | `is_not_empty`                     | Checks whether the values in the input column are not empty (but may be null).                                                                                                                                                                                                                                                                                                                                                    | `column`: column to check (can be a string column name or a column expression)                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (mwojtyczka) [inline docs/dqx/docs/reference/quality_checks.mdx:32]: for consistency, we should have `trim_strings` in this check as well

Comment (ghanse) [reply docs/dqx/docs/reference/quality_checks.mdx:32]: Added a `trim_strings` argument for the following checks:
- `is_not_empty`
- `is_empty`
- `is_not_null_and_not_empty`
- `is_null_or_empty`

File: docs/dqx/docs/reference/quality_checks.mdx (hunk 4 above)
Comment (ghanse) [reply docs/dqx/docs/reference/quality_checks.mdx]: Updated the docs

File: tests/integration/test_row_checks.py [hunk 6]
```diff
@@ -160,6 +163,139 @@ def test_col_is_not_null(spark):
...
+        [
+            ["str1", 1, {"val": "a"}, ["a", "b"]],
+            ["", None, {"val": ""}, [None, "a"]],
+            [" ", 3, {"val": None}, ["", "a"]],
+        ],
+        input_schema,
+    )
+
+    actual = test_df.select(
+        is_empty("a"),
```
Comment (mwojtyczka) [inline tests/integration/test_row_checks.py:248]: pls add case with trim_strings

**Review by mwojtyczka — APPROVED** [review]
LGTM
